cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java

```diff
@@ -170,4 +172,15 @@ public static Comparable convertFieldValue(String where, String kind, String fie
                       where, kind, stringValue, fieldName, fieldType.name()), e);
     }
   }
+
+  public static void addPartitions(Map<String, String> arguments, Iterator<Partition> partitionIterator) {
```

Comment:
Since it's just a wrapper method, the new method on PartitionedFileSetArguments isn't necessary.

The main change of this PR that's needed is delegating to the underlying files dataset for getInputFormatConfiguration, in case the PartitionFilter is missing.
https://github.com/caskdata/cdap/pull/3149/files#diff-3c5d2259d081358844e619de52b3741aR517